prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -146,17 +146,17 @@ class WorkflowEngine(
 
             nextEvent = updateBlock(header, retryToken, txn)
             context?.let { context.logger.info("Finished updateBlock for $reportId") }
-            val retryJson = nextEvent.retryToken?.toJSON()
+            val retryJson = nextEvent!!.retryToken?.toJSON()
             updateHeader(
                 header.task.reportId,
                 currentEventAction,
-                nextEvent.eventAction,
-                nextEvent.at,
+                nextEvent!!.eventAction,
+                nextEvent!!.at,
                 retryJson,
                 txn
             )
         }
-        queue.sendMessage(nextEvent) // Avoid race condition by doing after txn completes.
+        if (nextEvent != null)  queue.sendMessage(nextEvent!!) // Avoid race condition by doing after txn completes.
```

Comment:
Bug Fix:   This line was getting the Kotlin equiv of a NullPointerException, because the lateinit variable was not initialized if the "Weirdness" (above) occurs.    Its weirdness, but its not a bug/error.